modules/setting/setting.go

```diff
@@ -444,7 +444,19 @@ func getAppPath() (string, error) {
 	}
 	// Note: we don't use path.Dir here because it does not handle case
 	//	which path starts with two "/" in Windows: "//psf/Home/..."
-	return strings.ReplaceAll(appPath, "\\", "/"), err
+	appPath = strings.ReplaceAll(appPath, "\\", "/")
+
+	// Fix issue 16209
```

Comment:
Can you write more comments here? It will be easier for other people to understand what the issue is and how it is fixed.

Reply:
I added comments about the problem, and how it is fixed,

Reply:
Thank you, it's clear for me. I think we should be more careful here:

1. We should output a INFO level log to tell users that we are in SNAP and changed the AppPath. Or users may be confused by this behavior.
2. We should check whether the `current` path really exists and is valid.